File: back-end/model/DonXin.py
from sqlalchemy import NVARCHAR, Column, String, Date, Time, ForeignKey
from database import Base

class DonXin(Base):
    __tablename__ = 'DonXin'
    MaDon = Column(String(20), primary_key=True)
    MaNV = Column(String(20), ForeignKey('NhanVien.MaNV'))
    LoaiDon = Column(NVARCHAR(50))
    NgayBatDau = Column(Date)
    NgayKetThuc = Column(Date)
    LyDo = Column(NVARCHAR(255))
    TrangThai = Column(NVARCHAR(50))
    MaQL = Column(String(20), ForeignKey('QuanLy.MaQL'), nullable=True)
    NgayGui = Column(Date)
    NgayDuyet = Column(Date)
    GhiChu = Column(NVARCHAR(255))

    def to_dict(self):
        return {
            "MaDon": self.MaDon,
            "MaNV": self.MaNV,
            "LoaiDon": self.LoaiDon,
            "NgayBatDau": self.NgayBatDau.isoformat() if self.NgayBatDau else None,
            "NgayKetThuc": self.NgayKetThuc.isoformat() if self.NgayKetThuc else None,
            "LyDo": self.LyDo,
            "TrangThai": self.TrangThai,
            "MaQL": self.MaQL,
            "NgayGui": self.NgayGui.isoformat() if self.NgayGui else None,
            "NgayDuyet": self.NgayDuyet.isoformat() if self.NgayDuyet else None,
            "GhiChu": self.GhiChu,
        }

File: front-end/lib/models/donxin.dart
class DonXin {
  final String maDon;
  final String userId; // Server trả về key là "userId"
  final String loaiDon;
  final String lyDo;
  final String ngayBatDau;
  final String ngayKetThuc;
  final String trangThai;
  final String? hoTen; // Tên nhân viên
  final String? maQL;
  final String? ngayGui;
  final String? ngayDuyet;
  final String? ghiChu;

  DonXin({
    required this.maDon,
    required this.userId,
    required this.loaiDon,
    required this.lyDo,
    required this.ngayBatDau,
    required this.ngayKetThuc,
    required this.trangThai,
    this.hoTen,
    this.maQL,
    this.ngayGui,
    this.ngayDuyet,
    this.ghiChu,
  });
factory DonXin.fromJson(Map<String, dynamic> json) {
  return DonXin(
    maDon: json['MaDon'] ?? '',
    userId: json['MaNV'] ?? '',
    loaiDon: json['LoaiDon'] ?? '',
    lyDo: json['LyDo'] ?? '',
    ngayBatDau: json['NgayBatDau'] ?? '',
    ngayKetThuc: json['NgayKetThuc'] ?? '',
    trangThai: json['TrangThai'] ?? '',
    hoTen: json['HoTen'],
    maQL: json['MaQL'],
    ngayGui: json['NgayGui'],
    ngayDuyet: json['NgayDuyet'],
    ghiChu: json['GhiChu'],
  );
}



  Map<String, dynamic> toJson() {
    return {
      'maDon': maDon,
      'userId': userId,
      'loaiDon': loaiDon,
      'lyDo': lyDo,
      'ngayBatDau': ngayBatDau,
      'ngayKetThuc': ngayKetThuc,
      'trangThai': trangThai,
      'maQL': maQL,
      'ngayGui': ngayGui,
      'ngayDuyet': ngayDuyet,
      'ghiChu': ghiChu,
    };
  }
}